iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -317,6 +318,7 @@ static void free_transport_handle_data(MQTTTRANSPORT_HANDLE_DATA* transport_data
     if (transport_data->mqttClient != NULL)
     {
         mqtt_client_deinit(transport_data->mqttClient);
+        transport_data->mqttClient = NULL;
```

Comment:
@ewertons @jspaith can you guys verify this is correct? I believe we should be clearing it to `NULL` in the case that the `mqtt_client_deinit()` frees it.

Reply:
This isn't actually needed and we should remove this line from PR.  

At bottom of this function we have a `free(transport_data);` and nothing in this function will be mucking with the `mqttClient`.  We only really NULL stuff out if its part of a structure that has an ongoing life so later code knows not to mess with it.

Reply:
So the issue is, line 320 here will free the mqtt client but it will not set the pointer to `NULL`. The code then proceeds down to line 347 where it calls `DestroyXioTransport(transport_data);` which clears via `mqtt_client_clear_xio(transport_data->mqttClient);`. Since the pointer to the `mqtt_client` is not `NULL`, it derefs and segfaults.

Without it, here is a callstack which segfaults:
![image](https://user-images.githubusercontent.com/36710865/96807368-34335400-13cb-11eb-9ae9-3d0aee431bff.png)

Reply:
Ah - I didn't see that.  Please don't segfault:).  The code as you have it makes sense then.